File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/file/preparation/queues/CompletingFileQueueProvider.java
package com.lasthopesoftware.bluewater.client.playback.file.preparation.queues;

import com.lasthopesoftware.bluewater.client.browsing.items.media.files.ServiceFile;

import java.util.List;

public class CompletingFileQueueProvider implements IPositionedFileQueueProvider {

	@Override
	public IPositionedFileQueue provideQueue(List<ServiceFile> playlist, int startingAt) {
		return new CompletingPositionedFileQueue(QueueSplicers.getTruncatedList(playlist, startingAt));
	}

	@Override
	public boolean isRepeating() {
		return false;
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/file/preparation/queues/CompletingFileQueueProvider.kt
package com.lasthopesoftware.bluewater.client.playback.file.preparation.queues

import com.lasthopesoftware.bluewater.client.browsing.files.ServiceFile
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId

class CompletingFileQueueProvider : ProvidePositionedFileQueue {
	override val isRepeating: Boolean = false

	override fun provideQueue(libraryId: LibraryId, playlist: List<ServiceFile>, startingAt: Int): PositionedFileQueue {
		return CompletingPositionedFileQueue(libraryId, QueueSplicers.getTruncatedList(playlist, startingAt))
	}
}
